snare.py

```diff
@@ -257,6 +216,60 @@ def handle_request(self, request, payload):
         if content:
             response.write(content)
         yield from response.write_eof()
+    
+    @asyncio.coroutine
+    def parse_tanner_response(self, requested_name, detection):
+        content_type = None
+        content = None
+        status_code = 200
+        headers = {}
+
+        if detection['type'] == 1:
+            if requested_name == '/':
+                requested_name = self.run_args.index_page
+            try:
+                requested_name = unquote(requested_name)
+                file_name = self.meta[requested_name]['hash']
+                content_type = self.meta[requested_name]['content_type']
+            except KeyError:
+                status_code = 404
+            else:
+                path = os.path.join(self.dir, file_name)
+                if os.path.isfile(path):
+                    with open(path, 'rb') as fh:
+                        content = fh.read()
+                    if content_type:
+                        if 'text/html' in content_type:
+                            content = yield from self.handle_html_content(content)
+
+        elif detection['type'] == 2:
+            payload_content = detection['payload']
+            if payload_content['page']:
+                try:
+                    file_name = self.meta[payload_content['page']]['hash']
+                    content_type = self.meta[payload_content['page']]['content_type']
+                    page_path = os.path.join(self.dir, file_name)
+                    with open(page_path, encoding='utf-8') as p:
+                        content = p.read()
+                except KeyError:
+                    content = '<html><body></body></html>'
+                    content_type = 'text\html'
+
+                soup = BeautifulSoup(content, 'html.parser')
+                script_tag = soup.new_tag('div')
+                script_tag.append(BeautifulSoup(payload_content['value'], 'html.parser'))
+                soup.body.append(script_tag)
+                content = str(soup).encode()
+            else:
+                content_type = mimetypes.guess_type(payload_content['value'])[0]
+                content = payload_content['value'].encode('utf-8')
+
+            if 'headers' in payload_content:
+                headers =  payload_content['headers']
+        else:
+            status_code = payload_content['status_code']
```

Comment:
If we have e.g. 500 error on tanner side, why do we need to pass it to the end user? Maybe it's better to return current page?